contracts/evoting/types/election.go

```diff
@@ -255,6 +259,25 @@ func (p *PublicBulletinBoard) DeleteUser(userID string) bool {
 	return false
 }
 
+// PubShare represents a public share.
+type PubShare struct {
```

Comment:
I think we can re-work a bit the data structures.
Since every node must publish their pubshares with their given index, we could only have a slice of pubshares, where the index denotes the node's index, which is the same as the index of the pubshare. Like so we also don't need to store the public key because it is already stored in the roster.

Reply:
Regarding the public key, we store it in order to make sure that one node only sends one set of pubShares. Or could we could we do this some other way?

Reply:
If we assume that we keep ordering in the slice of pubshares according to the node index we know when a node already published its pubshare. For example, with the following slice:

```go
pubshares {
 nil,
 nil,
 {...},
 {...},
 nil,
}
```

We know that nodes 2 and 3 already published their pubshares.